internal/runtime/client/client.go

```diff
@@ -153,6 +163,107 @@ func (c *client) Unregister(extensionConfig *runtimev1.ExtensionConfig) error {
+		return errors.Wrap(err, "failed to aggregate responses")
+	}
+	return nil
+}
+
+func aggregateResponses(responses []runtimehooksv1.Response) error {
+	// TODO:(killianmuldoon) implement proper aggregation logic.
+	return nil
+}
+
+// CallExtension make the call to the extension with the given name.
+// The response object passed will be updated with the response of the call.
+// An error is returned if the extension is not compatible with the hook.
+// If the ExtensionHandler returns a response with `Status` set to `Failure` the function returns an error
+// and the response object is updated with the response received from the extension handler.
+//
+// FailurePolicy of the ExtensionHandler is used to handle errors that occur when performing the external call to the extension.
+// - If FailurePolicy is set to Ignore, the error is ignored and the response object is updated to be the default success response.
+// - If FailurePolicy is set to Fail, an error is returned and the response object may or may not be updated.
+// Nb. FailurePolicy does not affect the following kinds of errors:
+// - Internal errors. Examples: hooks is incompatible with ExtensionHandler, ExtensionHandler information is missing.
+// - Error when ExtensionHandler returns a response with `Status` set to `Failure`.
+func (c *client) CallExtension(ctx context.Context, hook catalog.Hook, name string, request runtime.Object, response runtimehooksv1.Response) error {
+	gvh, err := c.catalog.GroupVersionHook(hook)
+	if err != nil {
+		return errors.Wrap(err, "failed to compute GroupVersionHook")
+	}
+	registration, err := c.registry.Get(name)
+	if err != nil {
+		return errors.Wrapf(err, "failed to retrieve ExtensionHandler with name %q", name)
+	}
+	if gvh.GroupHook() != registration.GroupVersionHook.GroupHook() {
+		return fmt.Errorf("ExtensionHandler %q does not match group %s, hook %s", name, gvh.Group, gvh.Hook)
+	}
+	var timeoutDuration time.Duration
+	if registration.TimeoutSeconds != nil {
+		timeoutDuration = time.Duration(*registration.TimeoutSeconds) * time.Second
+	}
+	opts := &httpCallOptions{
+		catalog: c.catalog,
+		config:  registration.ClientConfig,
+		gvh:     registration.GroupVersionHook,
+		name:    strings.TrimSuffix(registration.Name, "."+registration.ExtensionConfigName),
+		timeout: timeoutDuration,
+	}
+	err = httpCall(ctx, request, response, opts)
+	if err != nil {
+		// If the error is errCallingExtensionHandler then apply failure policy to calculate
+		// the effective result of the operation.
+		ignore := *registration.FailurePolicy == runtimev1.FailurePolicyIgnore
+		if _, ok := err.(errCallingExtensionHandler); ok && ignore {
+			// Update the response to a default success response and return.
+			// - Set status to success
+			// - Set message to empty string
+			response.SetStatus(runtimehooksv1.ResponseStatusSuccess)
+			response.SetMessage("")
+			return nil
+		}
+		return errors.Wrap(err, "failed to call extension")
```

Comment:
```suggestion
		return errors.Wrap(err, "failed to call ExtensionHandler")
```

also use upper case in l.260. It's fine for names. lower case is just that the error concat with `:` looks nicer